signer/tests/integration/setup.rs

```diff
@@ -440,6 +415,27 @@ pub async fn backfill_bitcoin_blocks(db: &PgStore, rpc: &Client, chain_tip: &bit
     }
 }
 
+/// Fetch all block headers from bitcoin-core and store it in the database.
+pub async fn fetch_canonical_bitcoin_blockchain(db: &PgStore, rpc: &Client) {
+    let chain_tip_info = rpc.get_blockchain_info().unwrap();
+    let mut block_header = rpc
+        .get_block_header_info(&chain_tip_info.best_block_hash)
+        .unwrap();
+
+    // There are no non-coinbase transactions below this height.
+    while block_header.height as u64 >= regtest::MIN_BLOCKCHAIN_HEIGHT {
+        let parent_header_hash = block_header.previous_block_hash.unwrap();
+        let bitcoin_block = BitcoinBlock {
+            block_hash: block_header.hash.into(),
+            block_height: block_header.height as u64,
+            parent_hash: parent_header_hash.into(),
+        };
+
+        db.write_bitcoin_block(&bitcoin_block).await.unwrap();
+        block_header = rpc.get_block_header_info(&parent_header_hash).unwrap();
+    }
```

Comment:
nit: is this snippet doing something different or we can just call `backfill_bitcoin_blocks`?
```suggestion
    backfill_bitcoin_blocks(db, rpc, &chain_tip_info.best_block_hash).await
```

Reply:
Oh right, of course. Will update.

Reply:
Done in https://github.com/stacks-network/sbtc/pull/1406/commits/c7d18638e9733d402d3c47adfeb3c68351f49001